aws/sign.go

```diff
@@ -309,16 +311,10 @@ func canonicalHeaders(sortedHeaderNames []string, host string, hdr http.Header)
 // lowercase hexadecimal string.
 func payloadHash(req *http.Request, hasher hasher) (string, error) {
```

Comment:
As discussed in IRC, http.NewRequest wraps all body parameters in an ioutil.NopCloser, so this type assertion won't work :(